src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -79,10 +79,22 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         }
         codeWriter.emit(" {\n")
       } else if (anonymousTypeArguments != null) {
-        val supertype = if (superinterfaces.isNotEmpty()) superinterfaces[0] else superclass
-        codeWriter.emitCode("object : %T(", supertype)
-        codeWriter.emitCode(anonymousTypeArguments)
-        codeWriter.emit(") {\n")
+
+        codeWriter.emitCode("object :")
+        val supertype = if (superclass != ANY) {
+          listOf(CodeBlock.of(" %T(%L)", superclass, anonymousTypeArguments))
+        } else {
+          listOf()
+        }
+
+        val allSuperTypes = supertype + superinterfaces.map { CodeBlock.of(" %T", it) }
+        if (allSuperTypes.size > 1) {
+          allSuperTypes.dropLast(1).forEach { codeWriter.emitCode(it).emit(",") }
+        }
+        if (allSuperTypes.isNotEmpty()) {
+          codeWriter.emitCode(allSuperTypes.last())
+        }
```

Comment:
These `if`s should be able to be replaced by the `joinToCode` helper.

Reply:
👍 thanks - I knew there must have been a better way!

Reply:
Much neater - thanks!